Mage.Sets/src/mage/cards/a/AlaundoTheSeer.java

```diff
@@ -0,0 +1,235 @@
+import mage.players.Player;
+import mage.target.TargetCard;
+import mage.target.common.TargetCardInHand;
+import mage.target.targetpointer.FixedTarget;
+import mage.util.CardUtil;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.keyword.HasteAbility;
+
+/**
+ *
+ * @author Zelane
+ */
+public final class AlaundoTheSeer extends CardImpl {
+
+    public AlaundoTheSeer(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[] { CardType.CREATURE }, "{2}{G}{U}");
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SHAMAN);
+
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(5);
+
+        Ability ability = new SimpleActivatedAbility(new AlaundoTheSeerEffect(), new TapSourceCost());
+        this.addAbility(ability);
+    }
+
+    private AlaundoTheSeer(final AlaundoTheSeer card) {
+        super(card);
+    }
+
+    @Override
+    public AlaundoTheSeer copy() {
+        return new AlaundoTheSeer(this);
+    }
+}
+
+class AlaundoTheSeerEffect extends OneShotEffect {
+
+    private static final FilterCard inExileFilter = new FilterCard();
+    static {
+        inExileFilter.add(TargetController.YOU.getOwnerPredicate());
+        inExileFilter.add(CounterType.TIME.getPredicate());
+    }
+
+    public AlaundoTheSeerEffect() {
+        super(Outcome.PutCardInPlay);
+        this.staticText = "Draw a card, then exile a card from your hand and put a number of time "
+                + "counters on it equal to its mana value. It gains \"When the last time counter "
+                + "is removed from this card, if it's exiled, you may cast it without paying its "
+                + "mana cost. If you cast a creature spell this way, it gains haste until end of "
+                + "turn.\" Then remove a time counter from each other card you own in exile.";
+    }
+
+    private AlaundoTheSeerEffect(final AlaundoTheSeerEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public AlaundoTheSeerEffect copy() {
+        return new AlaundoTheSeerEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+
+        // Draw a card
+        controller.drawCards(1, source, game);
+        if (controller.getHand().isEmpty()) {
+            return false;
+        }
+
+        // then exile a card from your hand and put a number of time counters on it equal to its mana value
+        TargetCard target = new TargetCardInHand().withChooseHint("to exile");
+        controller.chooseTarget(outcome, controller.getHand(), target, source, game);
+        Card card = game.getCard(target.getFirstTarget());
+        if (card == null) {
+            return false;
```

Comment:
Are you sure it's correct to return false if no card is exiled? I would think the last part of the ability (removing a time counter from each other card) still needs to apply.

Reply:
Great point, I'll revisit this